components/StarSearch/Widgets/LotteryFactorWidget.tsx

```diff
@@ -1,21 +1,47 @@
+import { useEffect, useState } from "react";
+import { captureException } from "@sentry/nextjs";
 import LotteryFactorChart from "components/Repositories/LotteryFactorChart";
 import { useRepositoryLottoFactor } from "lib/hooks/api/useRepositoryLottoFactor";
+import { shortenUrl } from "lib/utils/shorten-url";
+
+const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL || "http://localhost:3000";
 
 interface LotteryFactorWidgetProps {
   repoName: string;
 }
 
 const LotteryFactorWidget = ({ repoName }: LotteryFactorWidgetProps) => {
+  const longUrl = new URL(`/s/${repoName}?utm=starsearch-workspaces`, BASE_URL).toString();
```

Comment:
I added a UTM to differentiate from a regular repository page short URL. Lemme know what you think.